src/pools/PoolRouter.sol

```diff
@@ -436,21 +415,18 @@ contract PoolRouter is Auth, Multicall, IPoolRouter, IPoolRouterGatewayHandler {
     //----------------------------------------------------------------------------------------------
 
     /// @inheritdoc IPoolRouterGatewayHandler
-    function registerAsset(AssetId assetId, string calldata name, string calldata symbol, uint8 decimals)
+    function registerAsset(AssetId assetId, uint8 decimals)
         external
         auth
     {
-        assetRegistry.registerAsset(assetId, name, symbol, decimals);
+        poolRegistry.registerAsset(assetId, decimals);
     }
 
     /// @inheritdoc IPoolRouterGatewayHandler
     function depositRequest(PoolId poolId, ShareClassId scId, bytes32 investor, AssetId depositAssetId, uint128 amount)
         external
         auth
     {
-        address pendingShareClassEscrow = escrow(poolId, scId, EscrowId.PendingShareClass);
-        assetRegistry.mint(pendingShareClassEscrow, depositAssetId.raw(), amount);
```

Comment:
Wow! this is a big concept change. Are we sure we do not want to differentiate between deployment & undeployment capital? Not sure, right now, if that information can be retrieved by the `ShareClassManager`

Reply:
I don't believe that is really relevant on the share class side. That can always be queried on the vaults side, by looking at the available balance of the escrow. I feel that is sufficient now.

Reply:
Cool!!